### ISSUE TITLE
[BUG] Rebuild loop after Volume degraded

### ISSUE THREAD

**Issue by proligde** [longhorn/longhorn#2067]
**Describe the bug**

Volume doesn't rebuild - stuck in rebuild loop: One of three nodes constantly loses on of the replica nodes in the GUI, starts rebuilding, fails, starts rebuilding, fails and so on. Sometimes the rebuild reaches 100% and then fails. Sometimes it fails before even starting.

the cluster has no known network issues and is in active use without any problems aside from the longhorn-filesystem not able to recover.

**To Reproduce**

I can't reproduce it: It's basically a default 3-replica cluster-layout including the nfs provisioner. I've set it up on another cluster as well, where it runs without problems.

**Expected behavior**
The rebuild succeeds and the volume is no longer marked as degraded.

**Log**
instance-manager-r (on the affected node)

```
[pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3] time="2020-12-09T13:41:17Z" level=info msg="Closing ssync server"
[pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3] time="2020-12-09T13:41:17Z" level=info msg="Done running ssync server for file volume-snap-f97b33e9-59c2-44dc-beb3-ecffb5c9d047.img at port 10023"
time="2020-12-09T13:41:17Z" level=info msg="Running ssync server for file volume-snap-f97b33e9-59c2-44dc-beb3-ecffb5c9d047.img.meta at port 10024"
time="2020-12-09T13:41:17Z" level=info msg="Creating Ssync service"
time="2020-12-09T13:41:17Z" level=info msg="open: receiving fileSize: 178, setting up directIo: false"
time="2020-12-09T13:41:17Z" level=info msg="Ssync server opened and ready"
[pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3] time="2020-12-09T13:41:17Z" level=info msg="Sync agent gRPC server finished rebuilding replica/sync files for replica 10.42.0.136"
time="2020-12-09T13:41:17Z" level=info msg="Reloading volume"
time="2020-12-09T13:41:17Z" level=info msg="Closing ssync server"
time="2020-12-09T13:41:17Z" level=info msg="Done running ssync server for file volume-snap-f97b33e9-59c2-44dc-beb3-ecffb5c9d047.img.meta at port 10024"
[longhorn-instance-manager] time="2020-12-09T13:41:36Z" level=debug msg="Process update: pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-de3dd018: state error: Error: signal: killed"
[longhorn-instance-manager] time="2020-12-09T13:42:01Z" level=debug msg="Process Manager: start getting logs for process pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3"
[longhorn-instance-manager] time="2020-12-09T13:42:01Z" level=debug msg="Process Manager: got logs for process pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3"
```

longhorn-manager (on the affected node)
```
time="2020-12-09T13:28:36Z" level=debug msg="Instance pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3 starts running, IP 10.42.0.136"
time="2020-12-09T13:28:36Z" level=debug msg="Instance pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3 starts running, Port 10015"
time="2020-12-09T13:28:36Z" level=debug msg="Instance handler updated instance pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3 state, old state stopped, new state running"
time="2020-12-09T13:42:01Z" level=warning msg="Try to get requested log for pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3 on node chameleon-3"
time="2020-12-09T13:42:01Z" level=warning msg="pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3: time=\"2020-12-09T13:28:36Z\" level=info msg=\"Creating volume /host/mnt/storage/longhorn/replicas/pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-72d9aa44, size 1500017328128/512\""
time="2020-12-09T13:42:01Z" level=warning msg="pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3: time=\"2020-12-09T13:28:36Z\" level=info msg=\"Listening on data server 0.0.0.0:10016\""
time="2020-12-09T13:42:01Z" level=warning msg="pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3: time=\"2020-12-09T13:28:36Z\" level=info msg=\"Listening on sync agent server 0.0.0.0:10017\""
time="2020-12-09T13:42:01Z" level=warning msg="pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-e80390a3: time=\"2020-12-09T13:28:36Z\" level=info msg=\"Listening on gRPC Replica server 0.0.0.0:10015\""
```

the affected node does not mention any RAM or Disk Pressure. Still I removed other workloads to push the load under 2 but that didn't help either.

**Environment:**
 - Longhorn version: Current (as of 2020-12-09) rancher template 1.0.2
 - Kubernetes version: v1.15.4-rancher1-2 (I want longhorn to run before updating to 1.18/1.19)
 - Node config
   - OS Ubuntu 18.04
   - CPU per node: Bare Metal Intel(R) Xeon(R) CPU E3-1240 v3 @ 3.40GHz
   - Memory per node: 32GB
   - Disk type- 2TB SSD
   - Network bandwidth and latency between the nodes: 1GBit
 - Underlying Infrastructure Baremetal

**Comment by joshimoo** [comment]
@proligde Can you attach/email a support bundle, something seems to be killing the replica process `pvc-7df48764-b7db-4868-b9e2-f2e9d6ae73b6-r-de3dd018: state error: Error: signal: killed"`

That should allow us to investigate further also a dmesg from the node where this is happening would be useful.

**Comment by proligde** [comment]
@joshimoo Thanks for your request. I had looked into dmesg but there weren't any noticeable problems. "Unfortunately" the problem resolved itself after several hours in quite a mysterious way: Except of one failing node there now were two:

1st the previously failing Replica was shown as unavailable in the "Volume" Tab of the GUI (after klickin on the volume), although marked as "schedulable" and "Ready" in the "Node" Tab of the GUI.

2nd a new, 4th Replica (although 3 configured) showing "N/A" everywhere showed up in the "Volume" Tab.

Then I did the following:
- I went to the Node Tab -> "Edit Node and Disks" (of the problem node) -> Disable
- I went to the Volume Tab -> Deleted the problem Replica (I couldn't delete the N/A Replica)
- I rent back to the Node Tab --> "Edit Node and Disks" (of the problem node) -> Enabled again

Then somehow everything went back to normal: The N/A replica disappeared automatically, the problem Replica switched back to Running and the Replication process completed successfully. I've copied several hundred Gigs of data to the Volume since yesterday and it stays up and running with no more degradation whatsoever.

I'm a bit concerned about what caused the problem in the first place but since I can't reproduce it anymore I'm going to close this ticket.

Anyway: I had no dataloss whatsoever - great work guys! I used drbd, gluster and ceph before and although all these systems can't be compared to each other as they are too different this is a huge step forward in terms of easy-of-use for shared file systems.
